test/ArcxAnalyticsSdk.test.ts

```diff
@@ -89,290 +110,460 @@ describe('(unit) ArcxAnalyticsSdk', () => {
         })
-        expect(postRequestStub.getCall(0).calledWith(PROD_URL_BACKEND, '', '/identify')).to.be.true
-        expect(
-          postRequestStub.getCall(1).calledWith(PROD_URL_BACKEND, '', '/submit-event', {
-            identityId: TEST_IDENTITY,
-            event: FIRST_PAGE_VISIT,
-            attributes: {
-              url: TEST_JSDOM_URL,
-            },
-          }),
-        ).to.be.true
+        expect(postRequestStub.getCall(0)).calledWithExactly(
+          DEFAULT_SDK_CONFIG.url,
+          '',
+          '/identify',
+        )
+        expect(postRequestStub.getCall(1)).calledWith(DEFAULT_SDK_CONFIG.url, '', '/submit-event', {
+          identityId: TEST_IDENTITY,
+          event: FIRST_PAGE_VISIT,
+          attributes: {
+            url: TEST_JSDOM_URL,
+          },
+        })
       })
 
       it('calls _reportCurrentWallet if trackWalletConnections is true', async () => {
-        // eslint-disable-next-line @typescript-eslint/no-extra-semi
-        ;(window as any).ethereum = new MockEthereum()
-
         const reportCurrentWalletStub = sinon.stub(
           ArcxAnalyticsSdk.prototype,
-          <any>'_reportCurrentWallet',
+          '_reportCurrentWallet' as any,
         )
         await ArcxAnalyticsSdk.init(TEST_API_KEY, { trackWalletConnections: true })
 
         expect(reportCurrentWalletStub.calledOnce).to.be.true
       })
 
-      it('trackTransactions', async () => {
-        const transactionParams = {
-          gas: '0x22719',
-          from: '0xbb4153b55a59cc8bde72550b0cf16781b08ef7b0',
-          to: '0x03cddc9c7fad4b6848d6741b0ef381470bc675cd',
-          data: '0x97b4d89f0...082ec95a',
-        }
-        window.web3 = {
-          currentProvider: sinon.createStubInstance(MetaMaskInpageProvider),
-        }
-        const nonce = 12
-        ;(window.web3.currentProvider.request as sinon.SinonStub).resolves(nonce).withArgs({
-          method: 'eth_getTransactionCount',
-          params: [transactionParams.from, 'latest'],
-        })
-        await ArcxAnalyticsSdk.init('', {
-          ...ALL_FALSE_CONFIG,
-          trackTransactions: true,
-        })
-        expect(postRequestStub.getCall(0).calledWith(PROD_URL_BACKEND, '', '/identify')).to.be.true
+      it('registers a listener on event accountsChanged if trackWalletConnections is true', async () => {
+        await ArcxAnalyticsSdk.init(TEST_API_KEY, { trackWalletConnections: true })
 
-        await window.web3.currentProvider.request({
-          method: 'eth_sendTransaction',
-          params: [transactionParams],
-        })
-        expect(
-          postRequestStub
-            .getCall(1)
-            .calledWith(
-              PROD_URL_BACKEND,
-              '',
-              '/submit-event',
-              getAnalyticsData(CAUGHT_TRANSACTION_EVENT, { ...transactionParams, nonce }),
-            ),
-        ).to.be.true
+        expect(window.ethereum?.on).calledWithMatch('accountsChanged')
+      })
```

Comment:
What is the point of splitting it to two separate tests?

Reply:
Merged into 1